core/stream/format.go

```diff
@@ -129,6 +129,21 @@ func (f *Format) HasColorComponent() bool {
 	return false
 }
 
+// HasSingleColorComponent returns a component if the format contains a single
+// color component, otherwise nil. See ColorChannels for channels considered colors.
+func (f *Format) HasSingleColorComponent() *Component {
```

Comment:
This signature confuses me. I would expect it to return a bool, iff there's a single color component. Instead it returns the first color component it finds.